atlasdb-cassandra/src/main/java/com/palantir/atlasdb/cassandra/CassandraAtlasDbFactory.java

```diff
@@ -106,20 +101,23 @@ private static CassandraKeyValueServiceConfig toCassandraConfig(KeyValueServiceC
     @VisibleForTesting
     Refreshable<CassandraKeyValueServiceRuntimeConfig> preprocessKvsRuntimeConfig(
             Refreshable<Optional<KeyValueServiceRuntimeConfig>> runtimeConfig) {
-        return runtimeConfig.map(maybeConfig -> maybeConfig
-                .map(config -> {
-                    if (!(config instanceof CassandraKeyValueServiceRuntimeConfig)) {
-                        log.error(
-                                "Invalid KeyValueServiceRuntimeConfig. Expected a KeyValueServiceRuntimeConfig of"
-                                        + " type CassandraKeyValueServiceRuntimeConfig, found {}. Using latest valid"
-                                        + " CassandraKeyValueServiceRuntimeConfig.",
-                                SafeArg.of("configClass", config.getClass()));
-                        return latestValidRuntimeConfig;
-                    }
-                    latestValidRuntimeConfig = (CassandraKeyValueServiceRuntimeConfig) config;
-                    return latestValidRuntimeConfig;
-                })
-                .orElseGet(CassandraKeyValueServiceRuntimeConfig::getDefault));
+        Refreshable<KeyValueServiceRuntimeConfig> refreshableConfig = runtimeConfig.map(
+                maybeConfig -> maybeConfig.orElseGet(CassandraKeyValueServiceRuntimeConfig::getDefault));
+        return RefreshableWithInitialDefault.of(
+                refreshableConfig,
+                CassandraAtlasDbFactory::castOrThrow,
+                CassandraKeyValueServiceRuntimeConfig.getDefault());
+    }
+
+    private static CassandraKeyValueServiceRuntimeConfig castOrThrow(Object value) {
```

Comment:
instead of `Object`, maybe we should start with `KeyValueServiceRuntimeConfig` as the type for this param?